### ISSUE TITLE
Update to support webpack@2.2.0-rc.2 and upcoming "rc" versions

### ISSUE THREAD

**Issue by sompylasar** [MoOx/phenomic#928]
```
npm WARN phenomic@0.20.1 requires a peer of webpack@^1.11.0 || ^2.1.0-beta.25 but none was installed.
```

https://github.com/webpack/webpack/releases

**Comment by thangngoc89** [comment]
Hi there, could you send a PR to support webpack rc ?

**Comment by sompylasar** [comment]
Yes, but I don't think I will be able to fully test the setup. I think the maintainers did an awesome piece of work towards webpack2 that they know the best where it may fail and how to work that arount.

**Comment by MoOx** [comment]
Phenomic is already working with webpack 2.1.0-beta-25. Should cover the RC? Not a lot of changes since this version?

**Comment by sompylasar** [comment]
@MoOx I think should cover both, and yes, a lot of changes, see here: https://github.com/webpack/webpack/releases

For your convenience, copy-pasted below.

**v2.1.0-beta.26**
```
Breaking Changes:

-loader is no longer automatically added when resolving loaders

This was removed because it often lead to problems.

You can still opt-in to the old behavior with the resolveLoader.moduleExtensions configuration option.

Node.js 0.10 is no longer supported

Node.js LTS for this version is expired. You won't miss it.

Note: 0.12 will expire in December 2017. We will remove it soon too.

Features:

AMD require now supports an error callback as 3rd argument
Bugfixes:

Declarations from destruction now hides imports
The parser now detects identifiers in destruction and default arguments
The parser no longer crashes on empty patterns [, a]
The schema now validates more stuff correctly
--version now works correctly
```

**v2.1.0-beta.27**
```
Bugfixes:

fixed indexed access to imported namespace object import * as X; X["abc"]
Deprecated DedupePlugin
Display an additional message for the breaking change in beta.26 to hint the user
fixed schema validation problem with noParse
```

**v2.1.0-beta.28**
```
Major changes:

add import() as Code Splitting construct. It should be used instead of System.import when possible. System.import will be deprecated in webpack 2 release (removed in webpack 3) as it's behavior is incorrect according to the spec.
*.json files are now supported without the json-loader. You may still use it. It's not a breaking change.
webpack will now show performance warnings when bundle size is too big. You can disable them with performance.hints = false. It recommended to disable them in development but leave them enabled in production.
Bugfixes:

Generate equal code on each platform when using node.js modules, module, process, global, etc.
CLI: Fixes --module-bind for the -loader change
Validation: externals now allows boolean values
Loaders: Complex options are now correctly passed when using remainingRequest like with the style-loader
Fixes a event leak when using watch mode
Stats: exclude now only tests resource and not complete identifier
Fixes bug related to export * from ...
CLI: Fix behavior of --cache and --no-cache
It's now possible to combine webpack ES2015 modules with babel ES2015 modules
Features:

add output.strictModuleExceptionHandling option which handles exceptions in modules correctly, but with less performance. The more performant variant is enabled by default.
Template strings are now allowed at locations where expressions are allowed. import(`./template/${template}`)
Allow to match rules on resourceQuery
add __webpack_nonce__ to set nonce for chunk loading script tags
Improve tree shaking for import *
Performance:

Cache some expensive stuff
🎄 We are approaching the first Release Candidate...
```

**v2.2.0-rc.0**
```
This is the first RC for webpack 2. We will try to avoid breaking changes now until webpack 3.

Bugfixes:

id in generated code of a context module is correctly escaped
context modules are now affected by NamedModulesPlugin, HashedModuleIdsPlugin, DllPlugin
performance hints ignore SourceMaps now
don't include multiple use stricts
imports also mark the module as ES2015 module like exports
exit the process correctly when watching and a critical error happens
externals are no longer moved by the commons chunk plugin
Features:

entrypoints are flagged with [big] if they are over the limit
improve validation errors
allow to alias loaders to a value with query
performance improvements
```

**v2.2.0-rc.1**
```
Bugfixes:

webpack was not able to recover after an error in a file occured in watch mode
It was not possible to set UglifyJs screw_ie8 option
```

**v2.2.0-rc.2**
```
Features:

Stats output now disables only 15 modules by default (see new option stats.maxModules, --display-max-modules)
Stats for modules now includes a depth for modules which is the distance to the entry point.
add --display-depth for CLI to display the module depth in stats
Bugfixes:

Cached modules now correctly reset meta information
arrow functions in require.ensure now work correctly
```

**v2.2.0-rc.3**
```
Bugfixes:

fixes chunk loading in old IEs
fixes error which prevented using arrow functions in require.ensure (ccfc608)
fixes bug when passing function to rules.use (44baa7c)
fixes bug for --progress output (0636e7c)
fixes bug with export * and non-harmony modules (ee3d53c)
fixes bug when removing assets from the compilation (cb4c209)
```

**Comment by MoOx** [comment]
From what I see here is what we should look at:

- remove all DedupePlugin references if any (from my memory it's already commented for webpack 2 in our codebase)
- remove json-loader (not a breaking change tho)

No breaking changes should affect our codebase.

**Comment by sompylasar** [comment]
And, obviously, update the `peerDependencies`.

**Comment by MoOx** [comment]
Closed by 806da9a1e86af7259f85933ebec1483e19ebb39a
